internal/newusers/manager.go

```diff
@@ -33,40 +58,61 @@ type GroupInfo struct {
 }
 
 type options struct {
-	cache *cache.Cache
+	expirationDate  time.Time
+	cleanOnNew      bool
+	cleanupInterval time.Duration
+	procDir         string // This is to force failure in tests.
 }
 
 // Option is a function that allows changing some of the default behaviors of the manager.
 type Option func(*options)
 
-// WithCache sets the cache to use for the manager.
-func WithCache(c *cache.Cache) Option {
+// WithUserExpirationDate overrides the default time for when a user should be cleaned from the cache.
+func WithUserExpirationDate(date time.Time) Option {
 	return func(o *options) {
-		o.cache = c
+		o.expirationDate = date
 	}
 }
 
 // NewManager creates a new user manager.
-func NewManager(cacheDir string, args ...Option) (*Manager, error) {
-	opts := &options{}
+func NewManager(cacheDir string, args ...Option) (m *Manager, err error) {
+	opts := &options{
+		expirationDate:  time.Now().Add(-1 * defaultEntryExpiration),
+		cleanOnNew:      true,
+		cleanupInterval: defaultCleanupInterval,
+		procDir:         "/proc/",
+	}
 	for _, arg := range args {
 		arg(opts)
 	}
 
-	if opts.cache != nil {
-		return &Manager{cache: opts.cache}, nil
+	m = &Manager{
+		doClear:        make(chan struct{}),
+		quit:           make(chan struct{}),
+		cleanupQuitted: make(chan struct{}),
 	}
-
 	c, err := cache.New(cacheDir)
 	if err != nil {
 		return nil, err
 	}
+	m.cache = c
 
-	return &Manager{cache: c}, nil
+	if opts.cleanOnNew {
+		if activeUsers, err := getActiveUsers(opts.procDir); err != nil {
+			slog.Warn(fmt.Sprintf("Could not get list of active users: %v", err))
+		} else if err := m.cache.CleanExpiredUsers(activeUsers, opts.expirationDate); err != nil {
+			slog.Warn(fmt.Sprintf("Could not clean database %v", err))
```

Comment:
`slog.Warn` seems to support formatting, isn't it?

Reply:
It's a "bait". The value must be final.